CMakeLists.txt

```diff
@@ -1,6 +1,8 @@
 cmake_minimum_required(VERSION 3.0.2)
-
-if((${CMAKE_VERSION} VERSION_GREATER "3.9.0") OR (${CMAKE_VERSION} VERSION_EQUAL "3.9.0"))
+message(STATUS "CMake version '${CMAKE_VERSION}' using generator '${CMAKE_GENERATOR}'")
+if(((${CMAKE_GENERATOR} MATCHES "Visual Studio.*") OR (${CMAKE_GENERATOR} MATCHES "Xcode.*"))
```

Comment:
You left out "Unix Makefiles" and broke the Linux builds, it seems.

Reply:
Makefiles don't support generators, that's whe whole point

Reply:
Your latest commit broke it again for me. It also passed CI, can we find out what combination of cmake version / platform is not working?

Reply:
Hi Chris. I tested cmake 3.5.1 in Linux using Makefiles and Ninja and cmake 3.9.6 under CLion using Make and both work with my commit.